BridgeAppSDK/SBAKeychainWrapper.m

```diff
@@ -0,0 +1,317 @@
+        [query setObject:service forKey:(__bridge id)kSecAttrService];
+        [query setObject:key forKey:(__bridge id)kSecAttrAccount];
+#if !TARGET_IPHONE_SIMULATOR && defined(__IPHONE_OS_VERSION_MIN_REQUIRED)
+        if (accessGroup) {
+            [query setObject:accessGroup forKey:(__bridge id)kSecAttrAccessGroup];
+        }
+#endif
+        
+        OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, NULL);
+        if (status == errSecSuccess) {
+            if (data) {
+                NSMutableDictionary *attributesToUpdate = [[NSMutableDictionary alloc] init];
+                [attributesToUpdate setObject:data forKey:(__bridge id)kSecValueData];
+                
+                status = SecItemUpdate((__bridge CFDictionaryRef)query, (__bridge CFDictionaryRef)attributesToUpdate);
+                if (status != errSecSuccess) {
+                    if (error) {
+                        *error = [NSError errorWithDomain:NSOSStatusErrorDomain
+                                                     code:status
+                                                 userInfo:@{NSLocalizedDescriptionKey: [Localization localizedString:@"KEYCHAIN_UPDATE_ERROR_MESSAGE"]}];
+                    }
+                    returnValue = NO;
+                }
+            } else {
+                [self removeItemForKey:key service:service accessGroup:accessGroup error:error];
+            }
+        } else if (status == errSecItemNotFound) {
+            NSMutableDictionary *attributes = [[NSMutableDictionary alloc] init];
+            [attributes setObject:(__bridge id)kSecClassGenericPassword forKey:(__bridge id)kSecClass];
+            [attributes setObject:service forKey:(__bridge id)kSecAttrService];
+            [attributes setObject:key forKey:(__bridge id)kSecAttrAccount];
+#if TARGET_OS_IPHONE || (defined(MAC_OS_X_VERSION_10_9) && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9)
+            [attributes setObject:(__bridge id)kSecAttrAccessibleAfterFirstUnlock forKey:(__bridge id)kSecAttrAccessible];
+#endif
+            [attributes setObject:data forKey:(__bridge id)kSecValueData];
+#if !TARGET_IPHONE_SIMULATOR && defined(__IPHONE_OS_VERSION_MIN_REQUIRED)
+            if (accessGroup) {
+                [attributes setObject:accessGroup forKey:(__bridge id)kSecAttrAccessGroup];
+            }
+#endif
+            
+            status = SecItemAdd((__bridge CFDictionaryRef)attributes, NULL);
+            if (status != errSecSuccess) {
+                if (error) {
+                    *error = [NSError errorWithDomain:NSOSStatusErrorDomain
+                                                 code:status
+                                             userInfo:@{NSLocalizedDescriptionKey: [Localization localizedString:@"KEYCHAIN_ADD_ERROR_MESSAGE"]}];
+                }
+                returnValue = NO;
+            }
+        } else {
+            returnValue = NO;
```

Comment:
Again, guessing it's the same in RK, but if it finds the item with no error, it attempts to modify it, and hands back a descriptive error object if it fails; if the error is "item not found", it attempts to create it, again handing back an error object with details in case of problems; but if it's anything else, it just returns NO with no explanation. In other words, it seems like it would make sense to pass back the error details here as well, no? Or could this break something that depends on the current RK behavior of not setting the error value here?